### PR TITLE
Levelup from leveldown

### PR DESCRIPTION
**Pull request by ralphtheninja**
Re-creating https://github.com/Level/levelup/pull/299 as a new PR with cherry picked commits. Also using `isLevelDOWN()` from `abstract-leveldown`.

### CODE REVIEW

File: lib/levelup.js [hunk 1]
```diff
@@ -47,20 +47,30 @@ function LevelUP (location, options, callback) {
...
+    , down
 
   EventEmitter.call(this)
   this.setMaxListeners(Infinity)
 
-  if (typeof location == 'function') {
+  if (isLevelDOWN(location)) {
+    options = typeof options == 'object' ? options : {}
+    down = location
+    options.db = function () { return down }
```
Comment (juliangruber) [inline lib/levelup.js]: why is that still a function?  i think now is a good time to enable `options.db = down`

Comment (juliangruber) [inline lib/levelup.js]: ah I see in the tests that that already works, so we can simplify here?

**Comment by juliangruber** [comment]
given https://github.com/Level/levelup/pull/299#issuecomment-69700570, we should document that the leveldown must not have been opened

File: lib/levelup.js (hunk 1 above)
Comment (ralphtheninja) [inline lib/levelup.js]: Yes. This if-statement looks terrible :) Imo the best would be to remove the `location` parameter completely.

Comment (juliangruber) [inline lib/levelup.js]: Shall me make it only this:

```js
var db = levelup(leveldown(location))
```

Comment (ralphtheninja) [inline lib/levelup.js]: Skip the `callback` completely?

Comment (ralphtheninja) [inline lib/levelup.js]: I'm all for simplifying, but I'd prefer to do all of that in a separate PR. I guess this change will involve a new major?

**Comment by ralphtheninja** [comment]
Agree. I'll update README.

**Comment by ralphtheninja** [comment]
Locking PR until leveldown can keep track of its own open state.
